src/main/scala/util/gen.scala

```diff
@@ -275,18 +277,19 @@ object gen {
 
       def fromPhrase: Phrase => HostedModule =
         partialHostCompiler(name) composeWith
-          (hostFunDefToHostPart() x map(kernelDefToKernel()))
+          ((((x: FunDef) => x) x map(kernelDefToKernel())) andThen
+          hostFunDefToHostPart)
     }
 
-    private def hostFunDefToHostPart(gen: HostCodeGenerator =
-                                     shine.OpenCL.Compilation.HostCodeGenerator()
-                                    ): FunDef => CModule =
-      HostCodeModuleGenerator.funDefToModule(gen)
+    private val hostFunDefToHostPart:
+      ((FunDef, Seq[KernelModule])) => (CModule, Seq[KernelModule]) = { case (hm, kms) =>
+      ???
```

Comment:
What is going on here?

Reply:
The OpenCL host code generator now needs access to the dependent OpenCL kernel modules, so we cannot use the OpenCL host code generator for CUDA kernel modules without generalizing the implementation. We should probably have a generic host code generator and generic kernel modules as part of #106 . I can add a comment in the code.

Reply:
Yes sounds good. Happy to have this merged then.